src/olympia/amo/utils.py

```diff
@@ -360,10 +360,14 @@ def fetch_subscribed_newsletters(user_profile):
     return data['newsletters']
 
 
-def subscribe_newsletter(user_profile, basket_id):
-    response = basket.subscribe(user_profile.email, basket_id, sync='Y')
+def subscribe_newsletter(request, user_profile, basket_id):
```

Comment:
Since we handle the case where `request` is `None`, how about making it an optional keyword argument instead ? Makes it easier when debugging through a shell, too - something I've been doing a lot recently - you just need to pass the user and basket id.

Reply:
Fair point, I have no argument against it. I'll do that